### PR TITLE
[UT]Add ut for lifted codes

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Add ut for lifted codes.

**Which issue(s) this PR fixes**:
Part of  #2579 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
A flaking test record:

```log
  Begin Captured StdOut/StdErr Output >>
    cluster(member-e2e-dwx) is joined successfully
    member-e2e-dwx cluster cordoned
    I0928 07:22:14.645502   40327 deployment.go:155] The ResourceBinding(karmadatest-7kj24/deploy-78962-deployment) schedule result is: member-e2e-dwx,member1
    I0928 07:22:15.744636   40327 unjoin.go:296] Waiting for the cluster object member-e2e-dwx to be deleted
    I0928 07:22:16.746068   40327 unjoin.go:296] Waiting for the cluster object member-e2e-dwx to be deleted
    ...

  Begin Captured GinkgoWriter Output >>
    STEP: Creating cluster: member-e2e-dwx 09/28/22 07:21:27.1
    STEP: Joinning cluster: member-e2e-dwx 09/28/22 07:22:00.09
    STEP: wait cluster member-e2e-dwx ready 09/28/22 07:22:04.285
    STEP: cluster should have unschedulable:NoSchedule taint 09/28/22 07:22:09.471
    STEP: Creating PropagationPolicy(karmadatest-7kj24/deploy-78962) 09/28/22 07:22:09.496
    STEP: Creating Deployment(karmadatest-7kj24/deploy-78962) 09/28/22 07:22:09.556
    STEP: Removing Deployment(karmadatest-7kj24/deploy-78962) 09/28/22 07:22:14.648
    STEP: Removing PropagationPolicy(karmadatest-7kj24/deploy-78962) 09/28/22 07:22:14.653
    STEP: Unjoinning cluster: member-e2e-dwx 09/28/22 07:22:14.66
    STEP: Deleting clusters: member-e2e-dwx 09/28/22 07:22:45.761
  << End Captured GinkgoWriter Output

  Expected
      <[]string | len:2, cap:2>: ["member-e2e-dwx", "member1"]
  not to contain element matching
      <string>: member-e2e-dwx
  In [It] at: /home/runner/work/karmada/karmada/test/e2e/karmadactl_test.go:587

  Full Stack Trace
    github.com/karmada-io/karmada/test/e2e.glob..func11.3.2()
    	/home/runner/work/karmada/karmada/test/e2e/karmadactl_test.go:587 +0x2b4
```

https://github.com/karmada-io/karmada/actions/runs/3141420283/jobs/5103924737

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/hold cancel
/lgtm

**Comment by RainbowMango** [comment]
Let's hold for a while, this probably conflicts with #2554

File: pkg/util/lifted/nodeselector_test.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (jwcesign) [inline pkg/util/lifted/nodeselector_test.go:2]: shoud be 2022

File: pkg/util/lifted/nodeselector_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			panic(e)
+		}
+		return out
+	}
+	tc := []struct {
+		in        []corev1.NodeSelectorRequirement
+		out       labels.Selector
+		expectErr bool
+	}{
+		{in: nil, out: labels.Nothing()},
```
Comment (jwcesign) [inline pkg/util/lifted/nodeselector_test.go:51]: I prefer to give a name parameter and give a short description what it test, like 
```go
name: "Test if there is xxx, it works right"
```

File: pkg/util/lifted/nodeselector_test.go (hunk 1 above)
Comment (RainbowMango) [reply pkg/util/lifted/nodeselector_test.go:2]: We shouldn't update the copyright.

File: pkg/util/lifted/nodeselector_test.go (hunk 2 above)
Comment (RainbowMango) [reply pkg/util/lifted/nodeselector_test.go:51]: @Poor12 Any comments?
If this is the code copied from Kubernetes, we don't need to update it.

Comment (Poor12) [reply pkg/util/lifted/nodeselector_test.go:51]: Yeah, it's completely copied from Kubernetes. I guess unless we changed the code we donot need to change it.

**Review by RainbowMango — COMMENTED** [review]
/lgtm

**Pull request merged**